### DISCUSSION TITLE
Optimize 2d Lid Driven Cavity

### DISCUSSION THREAD

**Discussion by laplacedBen** [lululxvi/deepxde#1653] [Q&A]
Hi

I just started using deepxde, it seams very practical, thank for the grat work!
I now tried implementing the classic Benchmark Lid Driven Cavity

However I'm at a loss as how to optimmizse the parameters

I already tried several network. However I'd assume a diffrent Architecture like a CNN would be more applicaible for the problem at hand, could anyone point me in the direction how to implement a CNN with deepxde?

This is my current implementation:

`import numpy as np
import deepxde as dde
import matplotlib.pyplot as plt
import deepxde as dde
import pandas as pd

from deepxde.backend.set_default_backend import set_default_backend
from tensorflow.keras.callbacks import EarlyStopping
set_default_backend("pytorch")

def pde(x, y):
    u, v, p = y[:, 0:1], y[:, 1:2], y[:, 2:]
    u_x = dde.grad.jacobian(y, x, i=0, j=0)
    u_y = dde.grad.jacobian(y, x, i=0, j=1)
    v_x = dde.grad.jacobian(y, x, i=1, j=0)
    v_y = dde.grad.jacobian(y, x, i=1, j=1)
    p_x = dde.grad.jacobian(y, x, i=2, j=0)
    p_y = dde.grad.jacobian(y, x, i=2, j=1)
    u_xx = dde.grad.hessian(y, x, component=0, i=0, j=0)
    u_yy = dde.grad.hessian(y, x, component=0, i=1, j=1)
    v_xx = dde.grad.hessian(y, x, component=1, i=0, j=0)
    v_yy = dde.grad.hessian(y, x, component=1, i=1, j=1)
    continuity = u_x + v_y
    momentum_u = u*u_x + v*u_y + p_x - 0.01*(u_xx + u_yy)
    momentum_v = u*v_x + v*v_y + p_y - 0.01*(v_xx + v_yy)

    return [continuity, momentum_u, momentum_v]

# Define the geometry
geom = dde.geometry.Rectangle([0, 0], [1, 1])

bc_u_top = dde.DirichletBC(geom, lambda x: 1, lambda x, on_boundary: on_boundary and np.isclose(x[1], 1), component=0)

bc_u_walls = dde.DirichletBC(geom, lambda x: 0, lambda x, on_boundary: on_boundary and (np.isclose(x[1], 0) or np.isclose(x[0], 0) or np.isclose(x[0], 1)), component=0)

bc_v_walls = dde.DirichletBC(geom, lambda x: 0, lambda x, on_boundary: on_boundary, component=1)

bc = [bc_u_top, bc_u_walls, bc_v_walls]

# Define the PDE and boundary condition data
data = dde.data.PDE(geom, pde, bc, num_domain=2800, num_boundary=800)

# Define the neural network architecture

layer_size = [2] + [64] * 6 + [3]
activation = "tanh"
initializer = "Glorot uniform"

net = dde.maps.FNN(layer_size, activation, initializer)
# Define and compile the model
model = dde.Model(data, net)

model.compile("adam", lr=0.001)

losshistory, train_state = model.train(iterations=3000, display_every=10)

dde.optimizers.config.set_LBFGS_options(maxiter=15000, maxcor=100)
model.compile("L-BFGS-B")

# Train the model with early stopping
losshistory, train_state = model.train(iterations=5000, display_every=10)`

Also I want to know if there are some advises how to balance the number of points within the domain and on the boundry?

These are my current results
![centerline_velocities](https://github.com/lululxvi/deepxde/assets/77884145/1dfb0f9a-3ce9-4e9e-8f72-560db49b72b6)
![dc_2d_velocity](https://github.com/lululxvi/deepxde/assets/77884145/cb9c16d1-0e5a-452f-b69d-3faa63667432)
